Source/engine.cpp

```diff
@@ -1683,7 +1683,7 @@ void __fastcall Cl2DecodeFrm1(int x, int y, char *pCelBuff, int nCel, int width,
 		if (pCelBuff) {
 			if (nCel > 0) {
 				v9 = *(LONG*)&pCelBuff[4 * nCel];
-				pCelBuffa = v9;
+				pCelBuffa = (char *)v9;

```

Comment:
Ideally Cl2DecodeFrm1 should be cleaned up upstream, removing the need for this cast. But for now this should be ok.